MREUnityRuntime/MREUnityRuntimeLib/Core/Actor.cs

```diff
@@ -767,15 +767,10 @@ private void PatchAppearance(AppearancePatch appearance)
             else if (appearance.MaterialId != null)
             {
                 MaterialId = appearance.MaterialId.Value;
-                var sharedMat = MREAPI.AppsAPI.AssetCache.GetAsset(MaterialId) as Material;
-                if (sharedMat != null)
+                MREAPI.AppsAPI.AssetCache.OnCached(MaterialId, sharedMat =>
                 {
-                    Renderer.sharedMaterial = sharedMat;
-                }
-                else
-                {
-                    MREAPI.Logger.LogWarning($"Material {MaterialId} not found, cannot assign to actor {Id}");
-                }
+                    Renderer.sharedMaterial = (Material)sharedMat;
```

Comment:
>Renderer.sharedMaterial  [](start = 20, length = 24)

Same concern with order of deferred vs instant changes to a variable.

Reply:
Added preconditions.

---
In reply to: [306960857](https://github.com/microsoft/mixed-reality-extension-unity/pull/92#discussion_r306960857) [](ancestors = 306960857)